### PR TITLE
feat(components/lookup): add autocomplete test harness

### PR DESCRIPTION
**Pull request by Blackbaud-SandhyaRajasabeson**
[AB#2195354](https://dev.azure.com/blackbaud/Products/_boards/board/t/SKY%20UX%20Program/Stories/?workitem=2195354)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2789/storybook/)

Component Storybooks:

- _(no component storybooks affected in this pr)_ 

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2789/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2789/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2789/playground/)

File: apps/code-examples/src/app/code-examples/lookup/autocomplete/basic/demo.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    const searchResults = await harness.getSearchResults();
+    await expectAsync(searchResults[0].getDescriptorValue()).toBeResolvedTo(
+      'Blue',
+    );
+    await expectAsync(searchResults[0].getText()).toBeResolvedTo('Blue');
+
+    await searchResults[0].select();
+    const value = fixture.componentInstance.formGroup.controls['favoriteColor']
+      .value as { name: string };
```
Comment (Blackbaud-SteveBrush) [inline apps/code-examples/src/app/code-examples/lookup/autocomplete/basic/demo.component.spec.ts]: Writing `as` should throw up a "red flag" in most cases. In this scenario, it looks like the spec is complaining because the `FormGroup` is cast to an `any` type. You can solve this by creating a strictly-typed `FormGroup` in the demo component:
```
public formGroup = inject(FormBuilder).group({
  favoriteColor: new FormControl<{ name: string } | undefined>(undefined),
});
```

And then in the spec, you can retrieve the now-typed value:
```
const value =
      fixture.componentInstance.formGroup.get('favoriteColor')?.value;
```

File: libs/components/lookup/testing/src/autocomplete/autocomplete-harness.ts [hunk 2]
```diff
@@ -115,6 +121,29 @@ export class SkyAutocompleteHarness extends SkyComponentHarness {
...
 
+  /**
+   * Gets the text displayed when no search results are found.
+   */
+  public async getNoResultsFoundText(): Promise<string | undefined> {
+    const overlay = await this.#getOverlay();
+
+    if (!overlay) {
+      throw new Error(
+        'Cannot find no results found text as the dropdown is closed.',
```
Comment (Blackbaud-SteveBrush) [inline libs/components/lookup/testing/src/autocomplete/autocomplete-harness.ts]: Can we put quotes around the "no results found" to help the sentence read more clearly? `Cannot find "no results found" text because the dropdown is closed.`

File: libs/components/lookup/testing/src/autocomplete/autocomplete-harness.ts [hunk 3]
```diff
@@ -115,6 +121,29 @@ export class SkyAutocompleteHarness extends SkyComponentHarness {
...
+      );
+    }
+
+    try {
+      const noResultFoundText =
+        await overlay.querySelector('.sky-deemphasized');
+      return (await noResultFoundText?.text())?.trim();
+    } catch {
+      throw new Error(
+        'Cannot find no results found text as there are search results found.',
```
Comment (Blackbaud-SteveBrush) [inline libs/components/lookup/testing/src/autocomplete/autocomplete-harness.ts]: Same here; maybe reword it a bit:
```
Cannot find "no results found" text because there are search results.
```

File: libs/components/lookup/testing/src/autocomplete/autocomplete-harness.ts [hunk 4]
```diff
@@ -156,6 +185,7 @@ export class SkyAutocompleteHarness extends SkyComponentHarness {
    * to prevent consumers of the autocomplete harness from calling it.
    * The lookup harness, which extends the autocomplete harness, may
    * still access this feature, however.)
+   * @internal
```
Comment (Blackbaud-SteveBrush) [inline libs/components/lookup/testing/src/autocomplete/autocomplete-harness.ts:188]: Good thinking.

**Review by Blackbaud-SteveBrush — APPROVED** [review]
